### ISSUE TITLE
Strange behaviour in ordering

### ISSUE THREAD

**Issue by gagiuntoli** [protofire/solhint#362]
I have the following case:
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.15;

contract Example {
    uint256 public a;

    constructor() {
        a = 2;
    }

    function functionPublicView() public view returns (bool) {
        return true;
    }

    // solhint-disable-next-line ordering
    function functionPublic() public {
        a = 1;
    }

    function functionExternal() external returns (bool) {
        return true;
    }
}
```

With this configuration:

```json
{
  "extends": "solhint:recommended",
  "plugins": ["prettier"],
  "rules": {
    "code-complexity": ["error", 8],
    "compiler-version": ["error", ">=0.8.8"],
    "const-name-snakecase": "off",
    "constructor-syntax": "error",
    "func-visibility": ["error", { "ignoreConstructors": true }],
    "ordering": "warn",
    "max-line-length": ["error", 120],
    "not-rely-on-time": "off",
    "prettier/prettier": [
      "error",
      {
        "endOfLine": "auto"
      }
    ],
    "reason-string": ["warn", { "maxLength": 64 }]
  }
}
```

I would expect solhint to throw an error on the ordering. The external function is after the public ones. But actually, it doesn't throw anything (`solhint --config ./.solhint.json file.sol`).

Then, if I change in the order (also wrong):

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.15;

contract Example {
    uint256 public a;

    constructor() {
        a = 2;
    }

    // solhint-disable-next-line ordering
    function functionPublic() public {
        a = 1;
    }

    function functionPublicView() public view returns (bool) {
        return true;
    }

    function functionExternal() external returns (bool) {
        return true;
    }
}
```

it fails with the expected error:

```bash
contracts/v1/markets/Solhint.sol
  20:5  warning  Function order is incorrect, external  function can not go after public view function (line 16)  ordering

✖ 1 problem (0 errors, 1 warning)
```

**Comment by dbale-altoros** [comment]
@gagiuntoli 
Hello, sorry for the late response.
Seems like what is not working is the
`// solhint-disable-next-line ordering`
Even tried like this
`/* solhint-disable */
        function functionPublic() public {
            a = 1;
        }
/* solhint-enable */`

But it is still considering that function
I will close this issue an open a new one with this bug
